compiler/src/dotty/tools/dotc/ast/TreeInfo.scala

```diff
@@ -470,6 +477,14 @@ trait TypedTreeInfo extends TreeInfo[Type] { self: Trees.Instance[Type] =>
     case _ => false
   }
 
+  /** Is tree a compiler-generated `.apply` node that refers to the
+   *  apply of a function class? (implicit functions are excluded)
+   */
+  def isSyntheticApply(tree: Tree): Boolean = tree match {
+    case Select(qual, nme.apply) => tree.pos.end == qual.pos.end
```

Comment:
Yes, but even for macro-generated apply's I would assume that we should not eta-expand (since the eta-expansion would simply undo what was generated).